src/language-markdown/embed.js

```diff
@@ -57,7 +57,10 @@ function embed(path, print, textToDoc, options) {
     case "importExport":
       return textToDoc(node.value, { parser: "babel" });
     case "jsx":
-      return textToDoc(node.value, { parser: "__js_expression" });
+      return textToDoc(`<>${node.value}</>`, {
```

Comment:
```suggestion
      return textToDoc(`<$>${node.value}</$>`, {
```

`<$></$>` is not considered as `jsx` in `mdx`, so it will not break `</>` error location reports literally.

Reply:
@JounQin Can you expand on this? What's wrong with using a fragment here?

Reply:
@thorn0 Imagine original code `123</>456`, after wrapped it's `<>123</>456</>`, it's definitely invalid jsx, but babel's error location would be wrong even if we catch the exception and try to restore the original error location.

With `<$>123</>456</$>`, it's definitely clear where is the error location and we restore it by catching parsing error.

See my PR for detail.

Actually, if we've merged it three months ago, it will not be an issue like current one here.

Reply:
Why would we wrap `123</>456` in a fragment?

Reply:
So you mean, let's say we have this code:
```
<Test>123</>456</Test>
```
The MDX parser decides that it's block JSX and we pass it to Babel. Babel tries to parse it and says that this code is invalid.

Now let's suppose we replaced `<>...</>` with `<$>...</$>`. What has changed? If we parse this:
```
<Test>123</$>456</Test>
```
we'll find ourselves in the same situation.

Please tell me what I'm missing.

Reply:
`<Test>123</$>456</Test>` you are right here, but it is very uncommon to use `<$>` in `mdx` or `jsx`, it's a balance choose to me.

Reply:
I still think there isn't any issue here. If Prettier can't parse an embedded language block (e.g. JSX in MDX), it just skips this block and formats the rest of the file. The error you might see on the playground is playground-only. The user doesn't see it normally. So if an invalid JSX block is encountered, it simply stays unformatted. What is the problem?

Reply:
A correct error message is better for user to find output why they're not formatted as expected.

Besides, I'm the author of [eslint-mdx](https://github.com/rx-ts/eslint-mdx) which works very well with `eslint-plugin-prettier`, so the error location does make sense.

Reply:
The user doesn't see any error message about embedded languages in Prettier. What you see on the playground is debug-only information.

Reply:
I mean the error message from `eslint`.

Reply:
Does is get that error from Prettier?

Reply:
I'm not for sure, it emits from rule `prettier/prettier`.

But whatever, my point is that we can do better with correct error location, then why should we not?

Reply:
Because currently there is no error at all, so I don't understand what we're even talking about.

Reply:
But okay, this change is too simple to argue about it, I'll add it